rtengine/impulse_denoise.h

```diff
@@ -277,19 +257,15 @@ SSEFUNCTION void ImProcFunctions::impulse_nr (LabImage* lab, double thresh)
     }
 //now impulsive values have been corrected
 
-    for (int i = 0; i < height; i++) {
-        delete [] lpf[i];
-        delete [] impish[i];
-    }
-
-    delete [] lpf;
-    delete [] impish;
+    delete [] lpf[0];
+    delete [] impish[0];
```

Comment:
You still have two calls to `new`, so you should also have two calls to `delete`, i.e. think `delete [] lpf;` and friends are missing as an array of pointers also needs memory. Again, I would suggest a separate helper class that implicitly or explicitly converts to `float**` (or `T**` if you want to go template on it).